<?xml version="1.0"?>

<!--
Mon Oct 10 19:20:36 EEST 2016, Nikos Koukis

Launchfile demonstrating the mrpt_graphslam_2d application usages
-->

<launch>

    <arg name="robot_ns" value="youbot_15" />
    <arg name="output" default="screen" />
    <arg name="robot_marker_ID" value="mf1" />
    <arg name="disable_MRPT_visuals" default="false" />
    <arg name="bag_dir" value="$(find mrpt_graphslam_2d)/rosbags" />
    <arg name="bag_file" value="records_with_gt.bag"/>

    <!-- Should we also start an instance of rviz ? -->
    <arg name="start_rviz" default="true" />
    <arg name="rviz_file" value="$(find mrpt_graphslam_2d)/rviz/sr_graphslam_gt.rviz" />

    <param name="/use_sim_time" value="true" />

	<group ns="$(arg robot_ns)">
		<!-- Run single-robot graphSLAM -->
		<include file="$(find mrpt_graphslam_2d)/launch/graphslam.launch">
			<arg name="output" value="$(arg output)" />
            <arg name = "config_file" value = "$(find mrpt_graphslam_2d)/config/ros_odometry_2DRangeScans.ini"  />
            <!--<arg name = "config_file" value = "$(find mrpt_graphslam_2d)/config/ros_laser_odometry.ini"  />-->
            <!--<arg name = "config_file" value = "$(find mrpt_graphslam_2d)/config/ros_odometry_2DRangeScans_LC_version.ini"  />-->
			<arg name="anchor_frame_ID" value="$(arg robot_marker_ID)_anchor" />
            <arg name="NRD" value="CFixedIntervalsNRD" />
            <!--<arg name="NRD" value="CICPCriteriaNRD" />-->
            <arg name="ERD" value="CICPCriteriaERD" />
            <!--<arg name="ERD" value="CLoopCloserERD" />-->
			<arg name="GSO" value="CLevMarqGSO" />
			<arg name="play_own_rosbag" value="false" />
            <arg name="disable_MRPT_visuals" value="$(arg disable_MRPT_visuals)" />
		</include>
	</group>

	<node pkg="rosbag" type="play" name="rosbag_player" output="$(arg output)" args="--clock $(arg bag_dir)/$(arg bag_file)" />

	<!-- Start rviz -->
	<group if="$(arg start_rviz)">
		<node pkg="rviz" type="rviz" name="rviz_visualization" output="$(arg output)" args="-d $(arg rviz_file)" />
	</group>
</launch>
